x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_model_health.tsx

```diff
@@ -113,12 +115,14 @@ export interface TrainedModelHealthProps {
   modelState: TrainedModelState | MlModelDeploymentState;
   modelStateReason?: string;
   isDownloadable?: boolean;
+  WrapperElement?: React.FC<WrapperButtonProps>;
 }
 
 export const TrainedModelHealth: React.FC<TrainedModelHealthProps> = ({
   modelState,
   modelStateReason,
   isDownloadable,
+  WrapperElement,
```

Comment:
Rather than providing the WrapperElement as a prop here, I would suggest adding the `onClick` function as a prop here, and then providing en EmptyButton as a wrapper if that onClick is populated. That way you don't need to provide the WrapperElement through the parent, which leads to some awkward rendering.

Reply:
@sphilipse that really was a much simpler way to approach this, thank you! I've changed it to pass down just the function instead of the whole component.